test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,152 @@
+        // empty addressbook
+        assertViewErrorInvalidIndex(emptyAddressBook, emptyDisplayList, 1);
+
+        // non-empty addressbook
+        assertViewErrorInvalidIndex(addressBook, listWithAll, -1);
+        assertViewErrorInvalidIndex(addressBook, listWithAll, 0);
+        assertViewErrorInvalidIndex(addressBook, listWithAll, listWithAll.size() + 1);
+    }
+
+    @Test
+    public void viewCommand_personNotInAddressBook_returnsPersonNotInAddressBookMessage() throws Exception {
+        // generate person not in addressbook, add to displayList
+        Person someone = new Person(new Name("me"),
+                                    new Phone("123", true),
+                                    new Email("some@hey.go", true),
+                                    new Address("nus", false),
+                                    new UniqueTagList(Collections.emptySet()));
+        listWithAll.add(someone);
+
+        // empty addressbook
+        assertViewErrorPersonNotInAddressBook(emptyAddressBook, listWithAll, 1);
+
+        // non-empty addressbook
+        assertViewErrorPersonNotInAddressBook(addressBook, listWithAll, listWithAll.size());
+    }
+
+    @Test
+    public void viewCommand_validIndex_returnsPersonDetails() {
+        // person with no private information
+        assertViewSuccess(addressBook, listWithAll, 1);
+
+        // person with some private information
+        assertViewSuccess(addressBook, listWithAll, 2);
+
+        // addressbook has more people than displayList
+        assertViewSuccess(addressBook, listWithSome, 1);
+    }
+
+    /**
+     * Creates a new View command.
+     */
+    private Command generateViewCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        ViewCommand command = new ViewCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+    
+    /**
+     * Creates a new ViewAll command.
+     */
+    private Command generateViewAllCommand(AddressBook addressBook, List<ReadOnlyPerson> displayList, int index) {
+        Command command = new ViewAllCommand(index);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Asserts that the details of the person at specific index can be successfully retrieved
+     * and displayed.
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               list.get(index - 1).getAsTextHidePrivate());
+        assertViewCommandBehaviour(addressBook, list, expectedMessage, index);
+        
+        expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                        list.get(index - 1).getAsTextShowAll());
+        assertViewAllCommandBehaviour(addressBook, list, expectedMessage, index);
```

Comment:
By local, I meant inside the method. Reasons,
1. It's not good when a calculation is duplicated. 
2. its meaning is not immediately clear. 

Something like this will help clear both problems.

`int internalIndex = index - 1;  // because index is one-indexed`